code/modules/projectiles/magazines/rifles.dm

```diff
@@ -343,3 +343,38 @@
 	default_ammo = /datum/ammo/bullet/rifle/heavy
 	max_rounds = 30
 	icon_state_mini = "mag_rifle"
+
+//-------------------------------------------------------
+//TX29
+
+/obj/item/ammo_magazine/rifle/tx29
+	name = "\improper TX-29 magazine (10x24mm)"
+	desc = "A small capacity 10mm rifle magazine. Differs from a T-18 magazine enough to bypass relevant patents."
+	caliber = CALIBER_10X24_CASELESS
+	icon_state = "tx29"
+	w_class = WEIGHT_CLASS_NORMAL
+	default_ammo = /datum/ammo/bullet/rifle
+	max_rounds = 36
+	icon_state_mini = "mag_rifle_big"
+
+/obj/item/ammo_magazine/rifle/tx29launcher
+	name = "\improper TX-29 20mm airburst grenade magazine"
+	desc = "A 20mm magazine loaded with airburst grenades. For use with the TX-29."
+	caliber = CALIBER_20MM
+	icon_state = "tx29launcher_airburst"
+	w_class = WEIGHT_CLASS_NORMAL
+	default_ammo = /datum/ammo/tx29launcher
+	max_rounds = 5
+	icon_state_mini = "mag_sniper"
+
+/obj/item/ammo_magazine/rifle/tx29launcher/he
```

Comment:
![image](https://user-images.githubusercontent.com/6610922/156978856-e72156a3-e9b4-49a0-bffd-4700bfa66327.png)

Only two references of /obj/item/ammo_magazine/rifle/tx29launcher/he, which means it's not in /obj/item/storage/box/tx29_system nor in req.